model/proto/ip.proto

```diff
@@ -0,0 +1,29 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+syntax = "proto3";
+
+package elastic.apm.v1;
+
+option go_package = "github.com/elastic/apm-data/model/modelpb";
+
+message IP {
+  oneof ip_addr {
+    fixed32 v4 = 1;
+    bytes v6 = 2;
```

Comment:
**question**: did you measure the storage difference with the previous representation ? 

Also, I wonder if storing the ipv6 in two `fixed64` would have any advantage ?

Reply:
I've run the following against main:

```
	e := &modelpb.APMEvent{
		Host: &modelpb.Host{
			Ip: []string{"127.0.0.1"},
		},
	}
	c := codec.VTProto{}

	by, err := c.Encode(e)
	if err != nil {
		log.Panic(err)
	}

	fmt.Fprintf(os.Stdout, "%d", len(by))
```

The size is 14 bytes.

And against this branch:

```
	e := &modelpb.APMEvent{
		Host: &modelpb.Host{
			Ip: []*modelpb.IP{
				{
					IpAddr: &modelpb.IP_V4{V4: 2130706433},
				},
			},
		},
	}
	c := codec.VTProto{}

	by, err := c.Encode(e)
	if err != nil {
		log.Panic(err)
	}

	fmt.Fprintf(os.Stdout, "%d", len(by))
```

The size is 10 bytes.

Reply:
> Also, I wonder if storing the ipv6 in two fixed64 would have any advantage ?

The problem would be that we can't use `repeated` in a `oneoff` message. So we'd have to create a new message type, which may cause more overhead.

Reply:
OTOH we already have an extra layer due to the `oneof`. Would it be better to remove the `oneof` to avoid reflection? i.e.

```
message IP {
  fixed32 v4 = 1;
  bytes v6 = 2;
}
```

We could always put the low (or high) 4 bytes in V4, and the rest in V6. (Maybe field names would then be low/high (lo/hi)?)